libresin/libresin/utils/enum_mapper.hpp

```diff
@@ -0,0 +1,44 @@
+#ifndef RESIN_ENUM_MAPPER_HPP
+#define RESIN_ENUM_MAPPER_HPP
+
+#include <array>
+#include <cstddef>
+#include <string_view>
+#include <type_traits>
+
+namespace resin {
+const size_t kMaxEnumSize = 64;
+
+template <typename T>
+concept EnumWithCountConcept = requires {
+  std::is_enum_v<std::remove_cv_t<std::remove_reference_t<T>>>;
+  { T::_Count } -> std::convertible_to<T>;
+};
+
+template <EnumWithCountConcept EnumType>
+struct EnumMapping {
+  using enum_type = std::remove_cv_t<std::remove_reference_t<EnumType>>;
+
+  constexpr EnumMapping() = default;
+
+  template <size_t N>
+  constexpr explicit EnumMapping(const std::string_view (&input_names)[N]) {
```

Comment:
why are we using C-style arrays instead of C++ `std::array` here?

Reply:
I've change my mind, can we use `std::initializer_list` here?

Reply:
I am not sure if the `std::initializer_list`'s size will work with the `static_assertion` for count, I've tried but the gcc complained

Reply:
For `std::array` the compiler complains as it tries to convert the `std::initializer_list` to `std::array`.

Reply:
Okay then let's keep it as it is